spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala

```diff
@@ -151,6 +155,11 @@ class IcebergSparkSqlExtensionsParser(delegate: ParserInterface) extends ParserI
             isSnapshotRefDdl(normalized)))
   }
 
+  private def isIcebergProcedure(normalized: String): Boolean = {
+    // All builtin Iceberg procedures are under the 'system' namespace
+    SparkProcedures.names().asScala.map("system." + _).exists(normalized.contains)
```

Comment:
Do we have to do any casing here? Or are we always guaranteed to get lower casing?

Also minor nit: Maybe this should live within SparkProcedures? SparkProcedures.isIcebergProcedure(string)

Reply:
+1 to move this method to `SparkProcedures`.

Reply:
> Do we have to do any casing here? Or are we always guaranteed to get lower casing?

@RussellSpitzer the method accepts a "normalized" SQL here.

From the context, I think the idea here might be: use a simple rule to normalize the SQL and check if it statisfy the Iceberg syntax, 1) if yes, use Iceberg's parser to parse the **original**(not the normalized one, so all other things, like hint, comments, are preserved) SQL; 2) if not, leave it to the delegate parser

> Maybe this should live within SparkProcedures? SparkProcedures.isIcebergProcedure(string)

Given it only accepts a "normalized" SQL, I would prefer to leave it here, alongside with existing `isSnapshotRefDdl`.

Reply:
Ok got it, so Normalized is actually normalizedSql. My confusion is that this checks only for 50% of the proof that it is an Iceberg Procedure with the "call" check happening outside the function.

I would think we would want 
```java
isIcebergProcedure(normalized: String)  {
  normalized.startsWith("call") &&
        SparkProcedures.names().asScala.map("system." + _).exists(normalized.contains)
}
```

Reply:
@RussellSpitzer the `normalized.startsWith("call") &&` is applied on the caller side, I'm trying best to follow the exsiting `private def isIcebergCommand(sqlText: String): Boolean` to orginaze the code.

Reply:
@RussellSpitzer I adopted your suggestion.